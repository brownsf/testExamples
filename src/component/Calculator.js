/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import * as utils from '../Utils';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      options: [{ type: 'add', symbol: '+' }, { type: 'subtract', symbol: '-' }, { type: 'multiply', symbol: '*' }, { type: 'divide', symbol: '/' }],
    };
  }

  render() {
    const {
      first, second, result, options,
    } = this.state;
    const calc = (func) => {
      switch (func) {
        case 'add':
          this.setState({ result: utils.add(first, second) });
          break;

        default:
          break;
      }
    };
    return (
      <div className="calcContainer">
        <div>
          <h2>
Do some math,
            {this.props.name}
          </h2>
        </div>
        <div className="caclResult">{result}</div>
        <div className="calcNumbers">
          <label>Numbers</label>
          <div>
            <input
              data-test-id="first"
              value={first || ''}
              onChange={(e) => this.setState({ first: e.target.value })}
            />
          </div>

          <div>
            <input
              value={second || ''}
              onChange={(e) => this.setState({ second: e.target.value })}
            />
          </div>
        </div>
        <div className="caclFunctions">
          {options.map((opt) => (
            <button
              type="button"
              data-test-id={`button-${opt.type}`}
              key={opt.type}
              onClick={() => calc(opt.type)}
            >
              {opt.symbol}
            </button>
          ))}
        </div>
      </div>
    );
  }
}

Calculator.defaultProps = {
  name: 'Buddy',
};

export default Calculator;
