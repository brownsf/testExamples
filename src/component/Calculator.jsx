import React from "react";
import * as utils from "../Utils";

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      options: [{ type: "add", symbol: "+" }]
    };
  }

  render() {
    const { first, second, result } = this.state;
    const calc = func => {
      switch (func) {
        case "add":
          this.setState({ result: utils.add(first, second)});
          break;

        default:
          break;
      }
    };
    return (
      <div className="calcContainer">
        <div className="caclResult">{result}</div>
        <div className="calcNumbers">
          <label>Numbers</label>
          <div>
            <input
              value={first || ""}
              onChange={e => this.setState({ first: e.target.value })}
            />
          </div>

          <div>
            <input
              value={second || ""}
              onChange={e => this.setState({ second: e.target.value })}
            />
          </div>
        </div>
        <div className="caclFunctions">
          {this.state.options.map(opt => (
            <button
              key={opt.type}
              onClick={() => calc(opt.type)}
            >
              {opt.symbol}
            </button>
          ))}
        </div>
      </div>
    );
  }
}

export default Calculator;
