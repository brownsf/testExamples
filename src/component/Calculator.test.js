import React from 'react';
import { shallow, mount } from 'enzyme';
import Calculator from './Calculator';

describe('calculator', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Calculator />);
  });
  it('should render', () => {
    component = shallow(<Calculator />);
    expect(component).toMatchSnapshot();

    expect(component.find('input')).toHaveLength(2);
  });

  it('should run math calculations', () => {
    const wrapper = mount(<Calculator />);

    wrapper.setState({ first: 1, second: 2 });

    wrapper.find('button[data-test-id="button-add"]').simulate('click');
    expect(wrapper.state().result).toBe(3);
  });

  it('should update props', () => {
    const wrapper = mount(<Calculator />);

    wrapper.setProps({ name: 'Mike' });


    expect(wrapper).toMatchSnapshot();
  });
});
