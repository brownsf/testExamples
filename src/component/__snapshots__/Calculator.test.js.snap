// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator should render 1`] = `
<div
  className="calcContainer"
>
  <div>
    <h2>
      Do some math,
      Buddy
    </h2>
  </div>
  <div
    className="caclResult"
    data-testid="result"
  />
  <div
    className="calcNumbers"
  >
    <label>
      Numbers
    </label>
    <div>
      <input
        data-testid="first"
        onChange={[Function]}
        placeholder="First Number"
        value=""
      />
    </div>
    <div>
      <input
        data-testid="second"
        onChange={[Function]}
        placeholder="Second Number"
        value=""
      />
    </div>
  </div>
  <div
    className="caclFunctions"
  >
    <button
      data-testid="button-add"
      key="add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      data-testid="button-subtract"
      key="subtract"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      data-testid="button-multiply"
      key="multiply"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
    <button
      data-testid="button-divide"
      key="divide"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
</div>
`;

exports[`calculator should update props 1`] = `
<Calculator
  name="Mike"
>
  <div
    className="calcContainer"
  >
    <div>
      <h2>
        Do some math,
        Mike
      </h2>
    </div>
    <div
      className="caclResult"
      data-testid="result"
    />
    <div
      className="calcNumbers"
    >
      <label>
        Numbers
      </label>
      <div>
        <input
          data-testid="first"
          onChange={[Function]}
          placeholder="First Number"
          value=""
        />
      </div>
      <div>
        <input
          data-testid="second"
          onChange={[Function]}
          placeholder="Second Number"
          value=""
        />
      </div>
    </div>
    <div
      className="caclFunctions"
    >
      <button
        data-testid="button-add"
        key="add"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        data-testid="button-subtract"
        key="subtract"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        data-testid="button-multiply"
        key="multiply"
        onClick={[Function]}
        type="button"
      >
        *
      </button>
      <button
        data-testid="button-divide"
        key="divide"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
  </div>
</Calculator>
`;
